README.md

```diff
@@ -21,49 +73,409 @@ that's been made default from that version forward.
+gem 'webpacker', git: 'https://github.com/rails/webpacker.git'
 ```
 
-You can also see a list of available commands by running `./bin/rails webpacker`
+**Note:** Use `rake` instead of `rails` if you are using webpacker
+with rails version < 5.0
+
+
+## Integrations
+
+Webpacker by default ships with basic out-of-the-box integration
+for React, Angular, Vue and Elm. You can see a list of available
+commands/tasks by running `./bin/rails webpacker` or `./bin/rake webpacker`
+in rails version < 5.0
+
+
+### React
+
+To use Webpacker with [React](https://facebook.github.io/react/), create a
+new Rails app using new `--webpack=react` option:
+
+```bash
+# Rails 5.1+
+./bin/rails new myapp --webpack=react
+```
+
+(or run `./bin/rails webpacker:install:react` in a existing Rails app already
+setup with webpacker).
+
+The installer will add all relevant dependencies using yarn, any changes
+to the configuration files and an example React component to your
+project in `app/javascript/packs` so that you can experiment with React right away.
+
+
+### Angular with TypeScript
+
+To use Webpacker with [Angular](https://angularjs.org/), create a
+new Rails app using new `--webpack=angular` option:
+
+```bash
+# Rails 5.1+
+./bin/rails new myapp --webpack=angular
+```
+
+(or run `./bin/rails webpacker:install:angular` on a Rails app already
+setup with webpacker).
+
+The installer will add TypeScript and Angular core libraries using yarn plus
+any changes to the configuration files. An example component written in
+TypeScript will also be added to your project in `app/javascript` so that
+you can experiment with Angular right away.
+
+
+### Vue
+
+To use Webpacker with [Vue](https://vuejs.org/), create a
+new Rails app using new `--webpack=vue` option:
+
+```bash
+# Rails 5.1+
+./bin/rails new myapp --webpack=vue
+```
+(or run `./bin/rails webpacker:install:vue` on a Rails app already setup with webpacker).
+
+The installer will add Vue and required libraries using yarn plus
+any changes to the configuration files. An example component will
+also be added to your project in `app/javascript` so that you can
+experiment Vue right away.
+
+
+### Elm
+
+To use Webpacker with [Elm](http://elm-lang.org), create a
+new app using new `--webpack=elm` option:
+
+```
+./bin/rails new myapp --webpack=elm
+```
+
+(or run `./bin/rails webpacker:install:elm` on a Rails app already setup with webpacker).
+
+The Elm library and core packages will be added via Yarn and Elm itself.
+An example `Main.elm` app will also be added to your project in `app/javascript`
+so that you can experiment with Elm right away.
+
 
 ## Binstubs
 
 Webpacker ships with two binstubs: `./bin/webpack` and `./bin/webpack-dev-server`.
-They're thin wrappers around the standard webpack.js executable, just to ensure that the right configuration file is loaded depending on your environment.
+They're thin wrappers around the standard `webpack.js` and `webpack-dev-server.js`
```

Comment:
They're not really wrappers around `.js` files. They wrap each npm module's `bin/` script, e.g. `node_modules/.bin/webpack`.